tokio-stream/src/stream_ext.rs

```diff
@@ -1005,6 +1007,57 @@ pub trait StreamExt: Stream {
     {
         throttle(duration, self)
     }
+
+
+
+    /// Batches the items in the given stream using a maximum duration and size for each batch.
+    ///
+    /// This stream returns the next batch of items in the following situations:
+    ///  1. The inner stream has returned at least `max_size` many items since the last batch.
+    ///  2. The time since the first element of a batch is greater than the given duration.
+    ///  3. The end of the stream is reached
+    ///
+    /// The length of the returned vector is never empty or greater than capacity. Empty batches
+    /// will not be emitted if no items are received upstream.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// use std::time::Duration;
+    /// use tokio::time;
+    /// use tokio_stream::{self as stream, StreamExt};
+    /// use futures::FutureExt;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
```

Comment:
We can enable test this with this change.
```suggestion
    /// #[tokio::main]
    /// # async fn _unused() {}
    /// # #[tokio::main(flavor = "current_thread", start_paused = true)]
    /// async fn main() {
```

Reply:
i'm curious why 
```
   /// #[tokio::main]
    /// # async fn _unused() {}
```
is needed?

and how the test works with `start_paused = true` without using `time::advance`?

Reply:
The lines with a hashtag and a space are not shown in the documentation. I added the unused function to make the `#[tokio::main]` appear on the main function in the example, but have it use the `#[tokio::main(flavor = "current_thread", start_paused = true)]` in the actual test.

As for `time::advance`, that's because Tokio will auto-advance the timer to the next sleep if all tasks are idle.

Reply:
good to know, thanks!